xbmc/addons/GUIWindowAddonBrowser.cpp

```diff
@@ -184,14 +184,18 @@ bool CGUIWindowAddonBrowser::OnClick(int iItem, const std::string &player)
       g_mediaManager.GetNetworkLocations(shares);
       std::string path;
       if (CGUIDialogFileBrowser::ShowAndGetFile(shares, "*.zip", g_localizeStrings.Get(24041), path))
+      {
         CAddonInstaller::GetInstance().InstallFromZip(path);
+        Refresh();
```

Comment:
Nah, not needed then. The install happens in the background and it will message back and refresh for that after it's complete.